### PR TITLE
Testsuite: Allow for comparison operators in feature and compiler matches

### PR DESCRIPTION
**Pull request by tamiko**
In reference to #1412

Changes:

8df46c2 (Matthias Maier, 43 seconds ago)
   add a changes.h entry

9bd4bc1 (Matthias Maier, 5 minutes ago)
   Documentation: Update to reflect changes in version constraint handling

d09dc4b (Matthias Maier, 2 hours ago)
   Testsuite: Allow for comparison operators in feature and compiler matches

### CODE REVIEW

**Comment by tamiko** [comment]
@davydden Now it does...

**Comment by davydden** [comment]
@tamiko cool!

File: cmake/macros/macro_deal_ii_pickup_tests.cmake [hunk 1]
```diff
@@ -152,22 +152,59 @@ MACRO(DEAL_II_PICKUP_TESTS)
     #
 
     STRING(REGEX MATCHALL
-      "compiler=[^=]*=(on|off|yes|no|true|false)" _matches ${_test}
+      "compiler=([a-z]|[A-Z])*(>=|<=|=|>|<)(on|off|yes|no|true|false|[0-9]+(\\.[0-9]+)*)" _matches ${_test}
```
Comment (bangerth) [inline cmake/macros/macro_deal_ii_pickup_tests.cmake]: Maybe write `([a-z]|[A-Z])*` as `[a-zA-Z]*`? ;-)

**Comment by bangerth** [comment]
Nice, please merge after thinking about the regex.

I've thought about something like this in the past and thought that encoding things in the filename is really a bit clunky. A more elegant way would be to somehow encode this in a separate file. But for now the complexity of conditions we want to attach is moderate, so let's not worry about it.

File: doc/developers/testsuite.html [hunk 2]
```diff
@@ -455,14 +457,22 @@
...
-      <code>Intel</code>.
+      The first test will only be set up if the specified compiler string
+      (in this case <code>GNU</code>) is equal to the string
+      <code>${CMAKE_CXX_COMPILER_ID}</code>. Similarly, the second test
+      will be set up if both strings are different. Common compiler names
+      are <code>GNU</code>, <code>Clang</code>, or <code>Intel</code>.
+    </p>
+    <p>
+      The third declaration is an example of how to specify a version
+      constraint: This tets will be set up if deal.II is configured with
```
Comment (tjhei) [inline doc/developers/testsuite.html]: tets->test

File: doc/news/changes.h [hunk 3]
```diff
@@ -108,6 +108,14 @@ inconvenience this causes.
 
 
 <ol>
+  <li> Fixed: The testsuite now properly supports version constraints for
+  compilers and features. Those are annotated by
+  <code>.compiler=STRING(&lt;=|&gt;=|=|&lt;|&gt;)VERSION.</code>, or
```
Comment (tjhei) [inline doc/news/changes.h]: is this the correct way to encode <,>,... for doxygen?

File: cmake/macros/macro_deal_ii_pickup_tests.cmake (hunk 1 above)
Comment (tamiko) [inline cmake/macros/macro_deal_ii_pickup_tests.cmake]: Ups. Forgot to update this regex...

File: doc/news/changes.h (hunk 3 above)
Comment (tamiko) [inline doc/news/changes.h]: @tjhei Yes, this works.

**Pull request merged**